group-coordinator/src/main/java/org/apache/kafka/coordinator/group/metrics/GroupCoordinatorMetricsShard.java

```diff
@@ -82,7 +82,7 @@ public TimelineGaugeCounter(TimelineLong timelineLong, AtomicLong atomicLong) {
     /**
      * Streams group size gauge counters keyed by the metric name.
      */
-    private final Map<StreamsGroupState, TimelineGaugeCounter> streamsGroupGauges;
+    private volatile Map<StreamsGroupState, Long> streamsGroupGauges;
```

Comment:
Why this change (`TimelineGaugeCounter` -> `Long`)? I imagine it's lighter-weight alternative

Reply:
I was mostly following a change from the consumer groups here, but indeed it looks more lightweight